src/components/Products/Products.jsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from "react";
+import axios from 'axios';
+import Pagination from "./Pagination/Pagination";
+import Product from "./Product/Product";
+
+const Products = () => {
+  const [products, setProducts] = useState([]);
+  const [totalProducts, setTotalProducts] = useState();
+  const [currentPage, setCurrentPage] = useState(1);
+  const [productsPerPage, setProductsPerPage] = useState(16);
+
+  const pages = [];
+  for(let i = 1; i < Math.ceil(totalProducts / productsPerPage); i++){
+    pages.push(i);
+  }
+
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const response = await axios.get(`http://localhost:3000/products?_page=${currentPage}&_limit=${productsPerPage}`);
+      const data = response.data;
+      setProducts(data);
+      setTotalProducts(response.headers['x-total-count']);
+    };
+
+    fetchData();
+  }, [currentPage]);
+
+  return (
+    <div>
+      <div className="d-flex flex-wrap">
+        {products && products.map(product => {
+          return (
+            <div className="col-md-3">
+              <Product key={product.objectID} product={product} />
+            </div>
+          )
+        })}
```

Comment:
Mấy cái tác vụ chỉ có return đơn giản như này thôi thì viết như này cho gọn anh nhé
```suggestion
        {products && products.map(product => (
            <div className="col-md-3">
              <Product key={product.objectID} product={product} />
            </div>
          ))}
```
Ngoài ra thì bổ sung key cho cái thằng cha bọc ngoài cùng nhé. Chứ k phải là key cho thằng con